Breaking change after png moved to codecs module

I received an issue that my crate fails to build when installing with cargo from crates.io.
https://github.com/okaneco/kmeans-colors/issues/35

I believe this was caused by #1328 which relocated png to the codecs module. This was a breaking change that prevents my crate from being compiled and installed. I have the `image` version in the Cargo.toml at `"0.23"`, the `image` version in Cargo.lock is at `"0.23.6"`. The change seems too large and non-backwards compatible to be included as a semver compatible minor or patch version. I had no idea it was broken until today.

## Expected

The crate should install successfully.

## Actual behaviour

The crate fails to compile and install.

```
   Compiling kmeans_colors v0.3.3
error[E0433]: failed to resolve: could not find `CompressionType` in `png`
  --> C:\Users\user\.cargo\registry\src\github.com-1ecc6299db9ec823\kmeans_colors-0.3.3\src\bin\kmeans_colors\utils.rs:89:25
   |
89 |             image::png::CompressionType::Best,
   |                         ^^^^^^^^^^^^^^^ could not find `CompressionType` in `png`

error[E0433]: failed to resolve: could not find `FilterType` in `png`
  --> C:\Users\user\.cargo\registry\src\github.com-1ecc6299db9ec823\kmeans_colors-0.3.3\src\bin\kmeans_colors\utils.rs:90:25
   |
90 |             image::png::FilterType::NoFilter,
   |                         ^^^^^^^^^^ could not find `FilterType` in `png`

error[E0433]: failed to resolve: could not find `CompressionType` in `png`
   --> C:\Users\user\.cargo\registry\src\github.com-1ecc6299db9ec823\kmeans_colors-0.3.3\src\bin\kmeans_colors\utils.rs:127:25
    |
127 |             image::png::CompressionType::Best,
    |                         ^^^^^^^^^^^^^^^ could not find `CompressionType` in `png`

error[E0433]: failed to resolve: could not find `FilterType` in `png`
   --> C:\Users\user\.cargo\registry\src\github.com-1ecc6299db9ec823\kmeans_colors-0.3.3\src\bin\kmeans_colors\utils.rs:128:25
    |
128 |             image::png::FilterType::NoFilter,
    |                         ^^^^^^^^^^ could not find `FilterType` in `png`

error: aborting due to 4 previous errors
```

## Reproduction steps

Try to install kmeans_colors using cargo.

```
cargo install kmeans_colors
```

The repository  
https://github.com/okaneco/kmeans-colors

The lines of code are located here  
https://github.com/okaneco/kmeans-colors/blob/e8543480c3173154b7bd4eab7ce22e89d8ae4570/src/bin/kmeans_colors/utils.rs#L89-L90  
https://github.com/okaneco/kmeans-colors/blob/e8543480c3173154b7bd4eab7ce22e89d8ae4570/src/bin/kmeans_colors/utils.rs#L127-L128

I can fix the issue by changing the fully qualified path.

```rust
image::codecs::png::CompressionType::Best,
image::codecs::png::FilterType::NoFilter,
//...
image::codecs::png::CompressionType::Best,
image::codecs::png::FilterType::NoFilter,
```

This broke identical code in another of my crates. https://github.com/okaneco/rscolorq
https://github.com/okaneco/rscolorq/blob/ce4205323ecbd45c3056678352780596cb497dc5/src/bin/rscolorq/utils.rs#L76-L77

The Cargo.lock has image `version = "0.23.10"`.

## Expected

The crate should install successfully.

## Actual behaviour

The crate fails to compile and install.

```rust
error[E0433]: failed to resolve: could not find `CompressionType` in `png`
  --> C:\Users\user\.cargo\registry\src\github.com-1ecc6299db9ec823\rscolorq-0.1.0\src\bin\rscolorq\utils.rs:76:21
   |
76 |         image::png::CompressionType::Best,
   |                     ^^^^^^^^^^^^^^^ could not find `CompressionType` in `png`

error[E0433]: failed to resolve: could not find `FilterType` in `png`
  --> C:\Users\user\.cargo\registry\src\github.com-1ecc6299db9ec823\rscolorq-0.1.0\src\bin\rscolorq\utils.rs:77:21
   |
77 |         image::png::FilterType::NoFilter,
   |                     ^^^^^^^^^^ could not find `FilterType` in `png`

error: aborting due to 2 previous errors
```

## Reproduction steps

Try to install rscolorq using cargo.

```
cargo install rscolorq
```

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -317,7 +317,8 @@ pub mod png {
     //! Decoding and Encoding of PNG Images
     #[allow(deprecated)]
     pub use crate::codecs::png::{
-        ApngDecoder, PNGEncoder, PNGReader, PngDecoder, PngEncoder, PngReader,
+        ApngDecoder, CompressionType, FilterType, PNGEncoder, PNGReader, PngDecoder, PngEncoder,
+        PngReader,
     };
 }
 #[cfg(feature = "pnm")]

